### PR TITLE
GHA frontend build and deploy to Azure blob

### PR DESCRIPTION
**Pull request by derekadombek**
# Pull Request

## Description
Automating the build process for the frontend to also deploy it to Azure blob storage. Eventually, we will need to drill down out deployment strategies for this and api layers

## Related Issues
[Link any related issues or tasks from your project management system.]

## Additional Notes
[Add any additional context or notes that reviewers should know about.]

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [x] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [x] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [x] I have updated the documentation, if applicable.
- [x] I have added or updated test cases to cover my changes, if applicable.
- [x] I have minimized the number of reviewers to include only those essential for the review.
- [x] I have notified teammates in the review thread to build awareness.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: .github/workflows/build-deploy-frontend.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      - name: Azure login
+        uses: azure/login@v2
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+      - name: Upload to Azure blob storage
+        shell: bash
+        run: |
+          az storage blob upload-batch --account-name reportvisionfrontenddev -d '$web' -s client-build/
```
Comment (zdeveloper) [inline .github/workflows/build-deploy-frontend.yml:69]: I am assuming the account name `reportvisionfrontenddev` will be parameterized at some point

**Review by zdeveloper — APPROVED** [review]
